File: algorithms/s_bubble.py
# The bubble sort makes multiple passes through a list. 

# It compares adjacent items and exchanges those that are out of order.
# Each pass through the list places the next largest value in its proper place. 

lst = [919, 15, 1, 4, 56, 32, 25, 85, 10, 3]

needSorting = True
i2 = 0

while needSorting :
    needSorting = False  
    for c in range (0, len(lst)):
        for i2 in range (0, len(lst) - 1 - c): # a sorted number is no more checked
            if lst[i2] > lst[i2+1]:                    
                lst[i2], lst[i2+1] = lst[i2+1], lst[i2] #bubble'd
                needSorting = True                          
                print("sorting in progress -> {}".format(lst)) # numbers moving the their right place
        
print("Sorted list : {}".format(lst))

File: Algorithms/s_bubble.js
// The bubble sort makes multiple passes through a list. 

// It compares adjacent items and exchanges those that are out of order.
// Each pass through the list places the next largest value in its proper place. 

// ------------------------------------
// I made this file when i was learning js; will convert it to a function later

var tab = [15, 1, 53, 5, 56, 32, 25, 919, 85, 6];
var isChanged = true;
var i2 = 0;

while (isChanged) {
    isChanged = false
    for ( c = 0; c <= tab.length; c++) {
        for (i2 = 0; i2 <= (tab.length-1-c); i2++) {
            if (tab[i2] < tab[i2+1]) {
                var tempX = tab[i2];
                var tempY = tab[i2+1];
                tab[i2] = tempY;
                tab[i2+1] = tempX;
                isChanged = true;
            }
        console.log("tab en cour de tri -> " + tab);
        }
    }
}
console.log("tab final -> " + tab)
